python/spear/env.py

```diff
@@ -100,28 +100,41 @@ def _request_launch_unreal_instance(self):
             return
 
         # write temp file
-        temp_config_file = os.path.realpath(os.path.join(self._config.SPEAR.TEMP_DIR, "config.yaml"))
+        temp_dir = os.path.realpath(os.path.join(self._config.SPEAR.TEMP_DIR))
+        temp_config_file = os.path.realpath(os.path.join(temp_dir, "config.yaml"))
 
         print("[SPEAR | env.py] Writing temp config file: " + temp_config_file)
 
-        if not os.path.exists(os.path.realpath(self._config.SPEAR.TEMP_DIR)):
-            os.makedirs(os.path.realpath(self._config.SPEAR.TEMP_DIR))
+        if not os.path.exists(temp_dir):
+            os.makedirs(temp_dir)
         with open(temp_config_file, "w") as output:
             self._config.dump(stream=output, default_flow_style=False)
 
         # create a symlink to SPEAR.DATA_DIR
         if self._config.SPEAR.DATA_DIR != "":
 
-            paks_dir = os.path.join(self._config.SPEAR.CONTENT_DIR, "Paks")
+            assert os.path.exists(self._config.SPEAR.DATA_DIR)
+
+            if sys.platform == "win32":
+                paks_dir = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(args.executable)), "..", "..", "Content", "Paks"))
+            elif sys.platform == "darwin":
+                paks_dir = os.path.realpath(os.path.join(args.executable, "Contents", "UE4", "SpearSim", "Content", "Paks"))
+            elif sys.platform == "linux":
+                paks_dir = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(args.executable)), "SpearSim", "Content", "Paks"))
+            else:
+                assert False
+
             assert os.path.exists(paks_dir)
+
+            # we don't use os.path.realpath here because we don't want to resolve the symlink
             spear_data_dir = os.path.join(paks_dir, "SpearData")
 
             if spear.path_exists(spear_data_dir):
                 print(f"[SPEAR | env.py] File or directory or symlink exists, removing: {spear_data_dir}")
                 spear.remove_path(spear_data_dir)
 
             print(f"[SPEAR | env.py] Creating symlink: {spear_data_dir} -> {self._config.SPEAR.DATA_DIR}")
-            os.symlink(self._config.SPEAR.DATA_DIR, spear_data_dir)
+            os.symlink(args.data_dir, spear_data_dir)
```

Comment:
This should be `self._config.SPEAR.DATA_DIR`, not `args.data_dir`.

Reply:
Oof. That's embarrassing. Good catch.